api/jobset/v1alpha2/jobset_webhook_test.go

```diff
@@ -747,3 +747,192 @@ func TestValidateCreate(t *testing.T) {
+			name: "update suspend",
+			js: &JobSet{
+				ObjectMeta: validObjectMeta,
+				Spec: JobSetSpec{
+					ReplicatedJobs: validReplicatedJobs,
+				},
+			},
+			oldJs: &JobSet{
+				ObjectMeta: validObjectMeta,
+				Spec: JobSetSpec{
+					Suspend:        ptr.To(true),
+					ReplicatedJobs: validReplicatedJobs,
+				},
+			},
+		},
+		{
+			name: "update labels",
+			js: &JobSet{
+				ObjectMeta: metav1.ObjectMeta{Name: "js", Labels: map[string]string{"hello": "world"}},
+				Spec: JobSetSpec{
+					ReplicatedJobs: validReplicatedJobs,
+				},
+			},
+			oldJs: &JobSet{
+				ObjectMeta: validObjectMeta,
+				Spec: JobSetSpec{
+					Suspend:        ptr.To(true),
+					ReplicatedJobs: validReplicatedJobs,
+				},
+			},
+		},
+		{
+			name: "forbid update on manged-by",
+			js: &JobSet{
+				ObjectMeta: metav1.ObjectMeta{Name: "js", Labels: map[string]string{LabelManagedBy: "world"}},
+				Spec: JobSetSpec{
+					ReplicatedJobs: validReplicatedJobs,
+				},
+			},
+			oldJs: &JobSet{
+				ObjectMeta: validObjectMeta,
+				Spec: JobSetSpec{
+					Suspend:        ptr.To(true),
+					ReplicatedJobs: validReplicatedJobs,
+				},
+			},
+			want: errors.Join(fmt.Errorf("metadata.labels[alpha.jobset.sigs.k8s.io/managed-by]: Invalid value: \"world\": field is immutable")),
+		},
+		{
+			name: "replicated jobs are mutable",
+			js: &JobSet{
+				ObjectMeta: validObjectMeta,
+				Spec: JobSetSpec{
+					ReplicatedJobs: []ReplicatedJob{
+						{
+							Name:     "test-jobset-replicated-job-0",
+							Replicas: 2,
+							Template: batchv1.JobTemplateSpec{
+								Spec: batchv1.JobSpec{
+									Parallelism: ptr.To[int32](2),
+								},
+							},
+						},
+						{
+							Name:     "test-jobset-replicated-job-1",
+							Replicas: 1,
+							Template: batchv1.JobTemplateSpec{
+								Spec: batchv1.JobSpec{
+									Parallelism: ptr.To[int32](1),
+								},
+							},
+						},
+					},
+				},
+			},
+			oldJs: &JobSet{
+				ObjectMeta: validObjectMeta,
+				Spec: JobSetSpec{
+					Suspend:        ptr.To(true),
+					ReplicatedJobs: validReplicatedJobs,
+				},
+			},
+		},
+		{
+			name: "parallelism and completetions are mutable",
+			js: &JobSet{
+				ObjectMeta: validObjectMeta,
+				Spec: JobSetSpec{
+					ReplicatedJobs: []ReplicatedJob{
+						{
+							Name:     "test-jobset-replicated-job-0",
+							Replicas: 1,
+							Template: batchv1.JobTemplateSpec{
+								Spec: batchv1.JobSpec{
+									Parallelism: ptr.To[int32](2),
+								},
+							},
+						},
+						{
+							Name:     "test-jobset-replicated-job-1",
+							Replicas: 1,
+							Template: batchv1.JobTemplateSpec{
+								Spec: batchv1.JobSpec{
+									Parallelism: ptr.To[int32](1),
+								},
+							},
+						},
+					},
+				},
+			},
+			oldJs: &JobSet{
+				ObjectMeta: validObjectMeta,
+				Spec: JobSetSpec{
+					Suspend:        ptr.To(true),
+					ReplicatedJobs: validReplicatedJobs,
+				},
+			},
+		},
+		{
+			name: "dropping a replicated job is valid",
```

Comment:
@ahg-g @danielvegamyhre added this here so we can discuss what kind of validation do we want on updates for replicated jobs.

I think this is probably not what we want..  OTOH if a jobset is suspended maybe someone can drop/add replicated jobs..